### PR TITLE
Add Qwen2Moe GGUF loading support 

### PR DESCRIPTION
**Pull request by VladOS95-cyber**
# What does this PR do?

Add Qwen2Moe GGUF loading support

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case. Link: https://github.com/huggingface/transformers/issues/33260
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?
Regarding the task @SunMarc @LysandreJik @ArthurZucker .

### CODE REVIEW

**Comment by VladOS95-cyber** [comment]
Hello @SunMarc @LysandreJik @ArthurZucker! I would like to ask a code review

File: src/transformers/integrations/ggml.py [hunk 1]
```diff
@@ -690,9 +725,15 @@ def converted(self) -> Tokenizer:
         return tokenizer
 
 
+class GGUFQwen2MoeConverter(GGUFQwen2Converter):
+    def __init__(self, tokenizer_dict):
+        super().__init__(tokenizer_dict)
+
+
```
Comment (SunMarc) [inline src/transformers/integrations/ggml.py:732]: Qwen2Moe and Qwen2 share the same tokenizer, we can just pass `GGUFQwen2Converter` in  `GGUF_TO_FAST_CONVERTERS`

**Review by SunMarc — APPROVED** [review]
Thanks for this clean PR @VladOS95-cyber ! LGTM !

**Comment by VladOS95-cyber** [comment]
> Thanks for this clean PR @vanpelt ! LGTM !

Why @vanpelt? =))

File: src/transformers/integrations/ggml.py (hunk 1 above)
Comment (VladOS95-cyber) [reply src/transformers/integrations/ggml.py:732]: Sure, will do

**Comment by SunMarc** [comment]
> Thank you for review! Why @vanpelt? =))
My bad ;) usually the first name github suggests me is the author of the PR

**Comment by VladOS95-cyber** [comment]
> > Thank you for review! Why @vanpelt? =))
> 
> My bad, editted ;) usually the first name github suggests me is the author of the PR

No worries at all :)

**Review by LysandreJik — APPROVED** [review]
Thanks @VladOS95-cyber!

**Comment by vanpelt** [comment]
Anytime guy's 🤪.  Nice work @VladOS95-cyber!

**Comment by SunMarc** [comment]
I'm hitting an error on our CI and locally for this test test_qwen2_moe_q4_0 @VladOS95-cyber . Can you have a look? 
Here's the traceback: 

```python
self = <ggml.test_ggml.GgufIntegrationTests testMethod=test_qwen2_moe_q4_0>

    def test_qwen2_moe_q4_0(self):
        tokenizer = AutoTokenizer.from_pretrained(self.qwen2_moe_model_id, gguf_file=self.q4_0_qwen2_moe_model_id)
        model = AutoModelForCausalLM.from_pretrained(
            self.qwen2_moe_model_id,
            gguf_file=self.q4_0_qwen2_moe_model_id,
            device_map="auto",
            torch_dtype=torch.float16,
        )

        text = tokenizer(self.example_text, return_tensors="pt").to(torch_device)
        out = model.generate(**text, max_new_tokens=10)

        EXPECTED_TEXT = "Hello everyone, I'm a newbie here and would like"
>       self.assertEqual(tokenizer.decode(out[0], skip_special_tokens=True), EXPECTED_TEXT)
E       AssertionError: 'Hello部分齐值得关注erc区域堪称 btnCancel跳舞�ASC' != "Hello everyone, I'm a newbie here and would like"
E       - Hello部分齐值得关注erc区域堪称 btnCancel跳舞�ASC
E       + Hello everyone, I'm a newbie here and would like

tests/quantization/ggml/test_ggml.py:359: AssertionError
```

Also here's some interesting logs: 

```
Some weights of Qwen2MoeForCausalLM were not initialized from the model checkpoint at RichardErkhov/Qwen_-_Qwen1.5-MoE-A2.7B-Chat-gguf and are newly initialized: ['model.layers.0.mlp.experts.0.down_proj.weight', 'model.layers.0.mlp.experts.0.gate_proj.weight', 'model.layers.0.mlp.experts.0.up_proj.weight', 'model.layers.0.mlp.experts.1.down_proj.weight', 'model.layers.0.mlp.experts.1.gate_proj.weight', 'model.layers.0.mlp.experts.1.up_proj.weight', 'model.layers.0.mlp.experts.10.down_proj.weight', 'model.layers.0.mlp.experts.10.gate_proj.weight', 'model.layers.0.mlp.experts.10.up_proj.weight', 'model.layers.0.mlp.experts.11.down_proj.weight', 'model.layers.0.mlp.experts.11.gate_proj.weight', 'model.layers.0.mlp.experts.11.up_proj.weight', 'model.layers.0.mlp.experts.12.down_proj.weight', 'model.layers.0.mlp.experts.12.gate_proj.weight', 'model.layers.0.mlp.experts.12.up_proj.weight', 
.....
```

Looks like we didn't manage to load the weights correctly. Also the GgufIntegrationTests.test_bloom_q8_0 test is also failing but it is easier to fix: 

```
>       self.assertEqual(tokenizer.decode(out[0], skip_special_tokens=True), EXPECTED_TEXT)
E       AssertionError: 'Hello, I just want to say that I am just' != 'Hello, I just want to say that I am very'
E       - Hello, I just want to say that I am just
E       ?                                     ^^^^
E       + Hello, I just want to say that I am very
E       ?                                     ^^^^

tests/quantization/ggml/test_ggml.py:422: AssertionError
```

**Comment by VladOS95-cyber** [comment]
@SunMarc sure, I'll take a look

**Pull request merged**